src/desugar.rs

```diff
@@ -1,206 +1,114 @@
-use crate::*;
+use crate::{proofs::RULE_PROOF_KEYWORD, *};
 
-pub(crate) type Fresh = dyn FnMut() -> Symbol;
+// TODO fix getting fresh names using modules
+const PROOF_UNDERSCORES: &str = "___";
```

Comment:
When we have modules, this will be fixed hopefully. Right now don't use three underscores in your egglog